examples/miscellaneous/plot_metadata_routing.py

```diff
@@ -205,42 +216,47 @@ def predict(self, X, **predict_params):
 # value}}}`` to pass to the underlying estimator's method. The ``object_name``
 # (``estimator`` in the above ``routed_params.estimator.fit`` example) is the
 # same as the one added in the ``get_metadata_routing``. ``validate_metadata``
-# makes sure all given metadata are requested to avoid silent bugs. Now, we
-# illustrate the different behaviors and notably the type of errors raised:
+# makes sure all given metadata are requested to avoid silent bugs.
+#
+# Next, we illustrate the different behaviors and notably the type of errors
+# raised.
 
-est = MetaClassifier(estimator=ExampleClassifier().set_fit_request(sample_weight=True))
-est.fit(X, y, sample_weight=my_weights)
+meta_est = MetaClassifier(
+    estimator=ExampleClassifier().set_fit_request(sample_weight=True)
+)
+meta_est.fit(X, y, sample_weight=my_weights)
 
 # %%
-# Note that the above example checks that ``sample_weight`` is correctly passed
-# to ``ExampleClassifier``, or else it would print that ``sample_weight`` is
+# Note that `ExampleClassifier` calling `check_metadata()` in the above
+# example checks that ``sample_weight`` is correctly passed to it. If it is
+# not, like in the following example, it would print that ``sample_weight`` is
 # ``None``:
 
-est.fit(X, y)
+meta_est.fit(X, y)
 
 # %%
 # If we pass an unknown metadata, an error is raised:
 try:
-    est.fit(X, y, test=my_weights)
+    meta_est.fit(X, y, test=my_weights)
 except TypeError as e:
     print(e)
 
 # %%
 # And if we pass a metadata which is not explicitly requested:
 try:
```

Comment:
Am I wrong, or is it that we will be shown this error message, also in the case where `test` is **not** requested metadata by **one** of the objects, instead of **not any**?

`MetaClassifier.fit got unexpected argument(s) {'test'}, which are not requested metadata in any object.`

I will check this later.

Reply:
Well, thinking about "not any" got be confused somehow. Now it's all clear.